metrics/sinks/statsd/driver.go

```diff
@@ -0,0 +1,193 @@
+		val, err := strconv.Atoi(opts["numMetricsPerMsg"][0])
+		if err != nil {
+			return config, fmt.Errorf("failed to parse `numMetricsPerMsg` field - %v", err)
+		}
+		config.numMetricsPerMsg = val
+	}
+	if len(opts["protocolType"]) >= 1 {
+		config.protocolType = strings.ToLower(opts["protocolType"][0])
+	}
+	if len(opts["prefix"]) >= 1 {
+		config.prefix = opts["prefix"][0]
+	}
+	if len(opts["renameLabels"]) >= 1 {
+		renameLabels := strings.Split(opts["renameLabels"][0], ",")
+		for _, renameLabel := range renameLabels {
+			kv := strings.SplitN(renameLabel, ":", 2)
+			config.renameLabels[kv[0]] = kv[1]
+		}
+	}
+	if len(opts["allowedLabels"]) >= 1 {
+		allowedLabels := strings.Split(opts["allowedLabels"][0], ",")
+		for _, allowedLabel := range allowedLabels {
+			config.allowedLabels[allowedLabel] = allowedLabel
+		}
+	}
+	labelStyle := DefaultLabelStyle
+	if len(opts["labelStyle"]) >= 1 {
+		switch opts["labelStyle"][0] {
+		case "lowerCamelCase":
+			labelStyle = SnakeToLowerCamel
+		case "upperCamelCase":
+			labelStyle = SnakeToUpperCamel
+		default:
+			glog.Errorf("invalid labelStyle - %s", opts["labelStyle"][0])
+		}
+	}
+	labelCustomizer := LabelCustomizer{config.renameLabels, labelStyle}
+	config.customizeLabel = labelCustomizer.Customize
+	glog.Infof("statsd metrics sink using configuration : %+v", config)
+	return config, nil
+}
+
+func (sink *statsdSink) ExportData(dataBatch *core.DataBatch) {
+	sink.Lock()
+	defer sink.Unlock()
+
+	var metrics []string
+	var tmpstr string
+	var err error
+	allowAllLabels := len(sink.config.allowedLabels) <= 0
+	for _, metricSet := range dataBatch.MetricSets {
+		var metricSetLabels map[string]string
+		if allowAllLabels {
+			metricSetLabels = metricSet.Labels
+		} else {
+			metricSetLabels = make(map[string]string)
+			for k, v := range metricSet.Labels {
+				_, allowed := sink.config.allowedLabels[k]
+				if allowed || allowAllLabels {
+					metricSetLabels[k] = v
+				}
+			}
+		}
+		for metricName, metricValue := range metricSet.MetricValues {
+			tmpstr, err = sink.formatter.Format(sink.config.prefix, metricName, metricSetLabels, sink.config.customizeLabel, metricValue)
+			if err != nil {
+				glog.Errorf("statsd metrics sink - failed to format metrics : %s", err.Error())
+				continue
+			}
+			metrics = append(metrics, tmpstr)
+		}
+		for _, metric := range metricSet.LabeledMetrics {
+			labels := make(map[string]string)
+			for k, v := range metricSetLabels {
+				labels[k] = v
+			}
+			for k, v := range metric.Labels {
+				_, allowed := sink.config.allowedLabels[k]
+				if allowed || allowAllLabels {
+					labels[k] = v
+				}
+			}
+			tmpstr, err = sink.formatter.Format(sink.config.prefix, metric.Name, labels, sink.config.customizeLabel, metric.MetricValue)
+			if err != nil {
+				glog.Errorf("statsd metrics sink - failed to format labeled metrics : %v", err)
+				continue
+			}
+			metrics = append(metrics, tmpstr)
+		}
+	}
+	if glog.V(4) {
```

Comment:
you can just `glog.V(4).Infof` -- this isn't in a hot loop

Reply:
Also, since you're dumping the entire metrics string, probably use a bit higher than 4